cgroup/cgroup_detect.go

```diff
@@ -0,0 +1,192 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+package cgroup
+
+// This entire file mostly copied from the datadog-agent
```

Comment:
Importing this package from the agent was a little difficult because most of these functions arent exported or intended for external use, so I copied them (and modified slightly to remove dd-agent specific config stuff)